### PR TITLE
Feature #28836 - allow multiple disassociating of provisioning templates

### PR DESCRIPTION
**Pull request by yifatmakias**

### CODE REVIEW

File: test/functional/associating_commands_test.rb [hunk 1]
```diff
@@ -93,14 +95,97 @@
...
+    end
+
+    it "should disassociate the given provisioning template-by id" do
+      params = ['--id=1', '--provisioning-template-id=2']
+
+      api_expects(:operatingsystems, :show).with_params({ "id": "1" }).returns(@os_before_update)
+
+      api_expects(:operatingsystems, :update) do |par|
+        par['id'] == '1' &&
+          par['operatingsystem'] == { 'provisioning_template_ids' => [] }
```
[the remark this answers is not in the document]
Comment (yifatmakias) [reply test/functional/associating_commands_test.rb]: You are right it should be 'provisioning_template_ids' => ['1'] in this case. I will fix it and also address the issue from the other PR you mentioned here.

**Comment by yifatmakias** [comment]
I thought that when disassociating a template it is more likely to want to delete it according to the full name or specific id so there will be no mistake and that's why I thought that search option here is not necessary. But if you think it should be added I will add it. What do you say?

**Comment by ofedoren** [comment]
@yifatmakias, I agree that search option has limited usage in this context, but if I wanted to disassociate all e.g. Kickstart templates, I'd rather use something like `--search ".*Kickstart.*"` instead of listing all of them.

**Comment by yifatmakias** [comment]
O.K, I will add the search option.
Thanks a lot for the review :)

**Comment by yifatmakias** [comment]
I agree that this problem is out of scope for this PR. I think we should create a new Redmine issue for this problem.

**Comment by ofedoren** [comment]
@yifatmakias, okay then :) I've created the issue: https://projects.theforeman.org/issues/29231.

So, merging this one, thanks!

**Pull request merged**